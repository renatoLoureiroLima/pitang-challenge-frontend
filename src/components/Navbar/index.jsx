import { Navbar } from 'react-bootstrap';
import { Link } from 'react-router-dom';

function index() {
  return (
    <div>
      <Navbar
        className='justify-content-center'
        bg='primary'
        variant='dark'
        data-testid='navbar-test'
      >
        <Link to='/'>
          <Navbar.Brand>Challenge Pitang</Navbar.Brand>
        </Link>
      </Navbar>
    </div>
  );
}

export default index;
