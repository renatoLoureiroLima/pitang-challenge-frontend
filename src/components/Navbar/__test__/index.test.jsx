import ReactDOM from 'react-dom';
import Navbar from '../index';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Navbar></Navbar>, div);
});

it('renders navbar correctly', () => {
  const { getByTestId } = render(<Navbar></Navbar>);
  expect(getByTestId('navbar-test')).toHaveTextContent('Challenge Pitang');
});

it('matches snapshot', () => {
  const tree = renderer.create(<Navbar></Navbar>).toJSON();
  expect(tree).toMatchSnapshot();
});
