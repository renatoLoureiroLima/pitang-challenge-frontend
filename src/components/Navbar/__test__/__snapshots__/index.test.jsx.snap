// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
    data-testid="navbar-test"
  >
    <span
      className="navbar-brand"
    >
      Challenge Pitang
    </span>
  </nav>
</div>
`;
